tests/conftest.py

```diff
@@ -488,13 +490,25 @@ def running_clients(hlwm, running_clients_num):
     return hlwm.create_clients(running_clients_num)
 
 
-@pytest.fixture()
-def x11():
-    from Xlib import X, display, Xutil, Xatom
+@pytest.fixture(scope="session")
+def x11_connection():
+    """ Long-lived fixture that maintains an open connection to the X11 display
+    for the entire duration of all tests. This avoids issues caused by Xvfb and
+    Xephyr resetting all properties whenever the last connection is closed. It
+    is probably a bit more efficient, too. """
+    display = Xlib.display.Display()
+    yield display
+    display.close()
+
 
+@pytest.fixture()
+def x11(x11_connection):
+    """ Short-lived fixture for interacting with the X11 display and creating
+    clients that are automatically destroyed at the end of each test. """
     class X11:
-        def __init__(self):
-            self.display = display.Display()
+        def __init__(self, x11_connection):
+            self.display = x11_connection.display
```

Comment:
This now becomes `self.display = x11_connection`

Reply:
Oops, thanks for fixing it